ivy/src/main/scala/sbt/IvyActions.scala

```diff
@@ -31,7 +31,7 @@ final class UpdateConfiguration(val retrieve: Option[RetrieveConfiguration], val
     logging: UpdateLogging.Value = this.logging): UpdateConfiguration =
     new UpdateConfiguration(retrieve, missingOk, logging)
 }
-final class RetrieveConfiguration(val retrieveDirectory: File, val outputPattern: String)
+final class RetrieveConfiguration(val retrieveDirectory: File, val outputPattern: String, val sync: Boolean)
```

Comment:
Given this is a binary incompatible change, you'll ahve to add a new constructor with the old patterns and instantiates a default sync value for boolean.